DBADashGUI/SchemaCompare/SchemaSnapshots.cs

```diff
@@ -77,10 +78,36 @@ private void GvSnapshotsDetail_CellContentClick(object sender, DataGridViewCellE
             {
                 ddlOld = Common.DDL((long)row["OldDDLID"]);
             }
-            var mode = e.ColumnIndex == colDiff.Index ? ViewMode.Diff : ViewMode.Code;
-            var frm = new Diff();
-            frm.SetText(ddlOld, ddl, mode);
-            frm.Show();
+
+            var diffToolBinaryPath = Properties.Settings.Default.DiffToolBinaryPath;
+            var diffToolArguments = Properties.Settings.Default.DiffToolArguments;
+
+            // If a custom diff tool is configured use that
+            if (!string.IsNullOrWhiteSpace(diffToolBinaryPath) && !string.IsNullOrWhiteSpace(diffToolArguments))
+            {
+                // TODO?: Check if binary path is valid? Try/catch on process start?
```

Comment:
If the tool doesn't exist it currently crashes.  It might be worth just adding a try/catch and displaying an error message.  This is worth it regardless of adding a check if the path is valid and might make the check redundant.  A path is valid check would need to work if the user has just specified the name of the app rather than the full path where they have added the app to PATH environment variable.